app/models/LiabilityCalculation/LiabilityCalculationResponse.scala

```diff
@@ -0,0 +1,174 @@
+ * limitations under the License.
+ */
+
+package models.LiabilityCalculation
+
+import play.api.libs.json._
+import play.api.libs.functional.syntax._
+import models.LiabilityCalculation.taxCalculation.TaxCalculation
+import models.LiabilityCalculation.taxDeductedAtSource.TaxDeductedAtSource
+
+sealed trait LiabilityCalculationResponseModel
+
+case class LiabilityCalculationError(status: Int, message: String) extends LiabilityCalculationResponseModel
+object LiabilityCalculationError {
+  implicit val format: OFormat[LiabilityCalculationError] = Json.format[LiabilityCalculationError]
+}
+
+case class LiabilityCalculationResponse(metadata: Metadata = Metadata(),
+                                        calculation: Calculation = Calculation()
+                                       ) extends LiabilityCalculationResponseModel
+object LiabilityCalculationResponse {
+  implicit val format: OFormat[LiabilityCalculationResponse] = Json.format[LiabilityCalculationResponse]
+}
+
+case class Metadata(calculationTimestamp: Option[String] = None, crystallised: Option[Boolean] = None)
+object Metadata {
+  implicit val writes: Writes[Metadata] = Json.writes[Metadata]
+
+  implicit val reads: Reads[Metadata] =
+    (
+      (JsPath \ "calculationTimestamp" ).readNullable[String] and
+        (JsPath \ "crystallised" ).readNullable[Boolean]
+      ) (Metadata.apply _)
+}
+
+case class Calculation(
+                        allowancesAndDeductions: AllowancesAndDeductions = AllowancesAndDeductions(),
+                        reliefs: Reliefs = Reliefs(),
+                        taxDeductedAtSource: TaxDeductedAtSource = TaxDeductedAtSource(),
+                        giftAid: GiftAid = GiftAid(),
+                        marriageAllowanceTransferredIn: MarriageAllowanceTransferredIn = MarriageAllowanceTransferredIn(),
+                        employmentAndPensionsIncome: EmploymentAndPensionsIncome = EmploymentAndPensionsIncome(),
+                        employmentExpenses: EmploymentExpenses = EmploymentExpenses(),
+                        stateBenefitsIncome: StateBenefitsIncome = StateBenefitsIncome(),
+                        shareSchemesIncome: ShareSchemesIncome = ShareSchemesIncome(),
+                        foreignIncome: ForeignIncome = ForeignIncome(),
+                        chargeableEventGainsIncome: ChargeableEventGainsIncome = ChargeableEventGainsIncome(),
+                        savingsAndGainsIncome: SavingsAndGainsIncome = SavingsAndGainsIncome(),
+                        dividendsIncome: DividendsIncome = DividendsIncome(),
+                        incomeSummaryTotals: IncomeSummaryTotals = IncomeSummaryTotals(),
+                        taxCalculation: TaxCalculation = TaxCalculation()
+                      )
+
+object Calculation {
+  implicit val writes: OWrites[Calculation] = Json.writes[Calculation]
+
+  implicit val reads: Reads[Calculation] =
+    (
+      (JsPath \ "allowancesAndDeductions").read[AllowancesAndDeductions] and
```

Comment:
We do not need custom reads here as its parameter names are equal to json field names.

Reply:
Good spot - you are right! I didn't realise it would read the non-native types automatically.